### PR TITLE
Feature - adjustsFontSizeToFit

### PR DESCRIPTION
**Pull request by MattFoley**

### CODE REVIEW

File: Libraries/Text/RCTText.m [hunk 1]
```diff
@@ -149,6 +165,101 @@ - (void)didMoveToWindow
...
+  return numberOfLines;
+}
+
+// Via Apple Text Layout Programming Guide
+//https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/TextLayout/Tasks/StringHeight.html
+- (CGSize)calculateSize:(NSTextStorage *)storage
+{
+  NSLayoutManager *layoutManager = [storage.layoutManagers firstObject];
+  NSTextContainer *textContainer = [layoutManager.textContainers firstObject];
+  (void) [layoutManager glyphRangeForTextContainer:textContainer];
```
Comment (jnhuynh) [inline Libraries/Text/RCTText.m]: what?

File: Libraries/Text/RCTText.m [hunk 2]
```diff
@@ -149,6 +165,101 @@ - (void)didMoveToWindow
...
+  return [layoutManager usedRectForTextContainer:textContainer].size;
+}
+
+//Start fresh with the original drawn string each time, in case frame has gotten larger
+- (void)resetDrawnTextStorage
+{
+  [_textStorage beginEditing];
+
+  NSRange originalRange = NSMakeRange(0, _originalString.length);
+  [_textStorage setAttributes:@{} range:originalRange];
```
Comment (jnhuynh) [inline Libraries/Text/RCTText.m]: Is there a case where `_textStorage` does have attributes? Are those attributed strings?

**Comment by jnhuynh** [comment]
LGTM

File: Libraries/Text/RCTText.m (hunk 2 above)
Comment (MattFoley) [inline Libraries/Text/RCTText.m]: So NSTextStorage is a subclass of NSAttributedString, we're purposefully fully wiping it here and replacing it with the original attributes. This way we start from scratch each time we want to autosize the text.

**Pull request merged**